### PR TITLE
Bug 1210252 - Migrate remaining mozL10n.DateTimeFormat use cases in System to mozIntl API

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1210252

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=11c52455bdd6b1095e467d9757a7af4d24ee02bf

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4ae20e82fcd19d57d733c79fa0d3208e58f79d57

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=aaa33281e93af5850f957d04ea569fd9d5d69bf2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ac610355e4bb4aaa2b536738505cdacd3766d4c1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dd5a5939054362f11c280776e6b31bc9da9ec63a

File: apps/system/test/unit/notification_screen_test.js [hunk 1]
```diff
@@ -31,7 +33,7 @@ var mocksForNotificationScreen = new MocksHelper([
   'SettingsListener',
   'SettingsURL',
   'Service',
-  'LazyLoader'
+  'LazyLoader',
```
Comment (albertopq) [inline apps/system/test/unit/notification_screen_test.js]: nit: let's remove this comma, or even better, use the the MocksHelper for loading the new mock

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3a10247977430ab6c42ad1353bdb55db702452c8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ab9beab28ccdc580987c964d3c3f01426254fbf5

File: apps/system/test/unit/notification_screen_test.js (hunk 1 above)
Comment (zbraniecki) [inline apps/system/test/unit/notification_screen_test.js]: I don't know how to do this. I tried adding "mozIntl" to the list but it didn't attach it on the `window`.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=82643cd733214419330f23fcf9bf99b1fd731dc9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3f47dd372b44d40683c7329890b934820f3fb9e8

**Pull request merged**